### PR TITLE
Use webdav instead of scp to copy files to the robot.

### PR DESCRIPTION
**Pull request by PeterJohnson**
Also combine a few ssh command steps.

This results in about a 2x speedup in deployment.

The jackrabbit logging is a bit noisy at present; looking into how to reduce it.

### CODE REVIEW

File: edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/build.xml [hunk 1]
```diff
@@ -184,28 +112,20 @@
...
-	   failonerror="false"
-       command=". /etc/profile.d/natinst-path.sh; chmod a+x ${deploy.dir}/${out}; ${deploy.kill.command};"/>
-
-    <sshexec host="${target}"
-         username="${username}"
-         password="${password}"
-         trust="true"
-         command="sync" />
+       failonerror="false"
+       command=". /etc/profile.d/natinst-path.sh; chown lvuser ${deploy.dir}/${out}; setcap 'cap_sys_nice=pe' ${deploy.dir}/${out}; chmod a+x ${deploy.dir}/${out}; ${deploy.kill.command}; sync"/>
```
Comment (calcmogul) [inline edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/build.xml:128]: Shouldn't this line have `setcap 'cap_sys_nice=pe' ${deploy.dir}/${out}`?

Comment (PeterJohnson) [reply edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/build.xml:128]: It does...

Comment (calcmogul) [reply edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/build.xml:128]: Oh, GitHub hid the horizontal scrollbar.

**Comment by AustinShalit** [comment]
Would this mean code deploys will happen over port 80?

**Comment by PeterJohnson** [comment]
Not entirely.  The deploy mostly happens over port 80, but there's still a couple of steps that require port 22.

**Comment by sciencewhiz** [comment]
I'm not able to connect. Any thoughts on debugging? I was able to connect using windows explorer over webdav.
```
deploy:
   [webdav] getting http://192.168.1.26/files/usr/local/frc/lib/WPI_Native_Libraries.properties
     [echo] Deploying libraries 
     [echo]  libntcore.so
   [webdav] putting C:\Users\Joe\wpilib\java\current\lib\native\lib\libntcore.so to http://192.168.1.26/files/tmp/libntcore.so
   [webdav] org.apache.http.NoHttpResponseException: 192.168.1.26:80 failed to respond
   [webdav] 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:141)
   [webdav] 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
   [webdav] 	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
   [webdav] 	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
   [webdav] 	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:165)
   [webdav] 	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
   [webdav] 	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
   [webdav] 	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
   [webdav] 	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
   [webdav] 	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
   [webdav] 	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
   [webdav] 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
   [webdav] 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)
   [webdav] 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:221)
   [webdav] 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:165)
   [webdav] 	at com.github.sardine.impl.SardineImpl.execute(SardineImpl.java:1036)
   [webdav] 	at com.github.sardine.impl.SardineImpl.execute(SardineImpl.java:1005)
   [webdav] 	at com.github.sardine.impl.SardineImpl.put(SardineImpl.java:916)
   [webdav] 	at com.github.sardine.impl.SardineImpl.put(SardineImpl.java:899)
   [webdav] 	at com.github.sardine.impl.SardineImpl.put(SardineImpl.java:887)
   [webdav] 	at com.github.sardine.impl.SardineImpl.put(SardineImpl.java:944)
   [webdav] 	at com.github.sardine.ant.command.Put.process(Put.java:113)
   [webdav] 	at com.github.sardine.ant.command.Put.execute(Put.java:74)
   [webdav] 	at com.github.sardine.ant.Command.executeCommand(Command.java:44)
   [webdav] 	at com.github.sardine.ant.SardineTask.execute(SardineTask.java:120)
   [webdav] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
   [webdav] 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
   [webdav] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   [webdav] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [webdav] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
   [webdav] 	at org.apache.tools.ant.Task.perform(Task.java:348)
   [webdav] 	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:68)
   [webdav] 	at net.sf.antcontrib.logic.IfTask.execute(IfTask.java:197)
   [webdav] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   [webdav] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   [webdav] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   [webdav] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [webdav] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
   [webdav] 	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:155)
   [webdav] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
   [webdav] 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
   [webdav] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   [webdav] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [webdav] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
   [webdav] 	at org.apache.tools.ant.Task.perform(Task.java:348)
   [webdav] 	at org.apache.tools.ant.taskdefs.Sequential.execute(Sequential.java:68)
   [webdav] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
   [webdav] 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
   [webdav] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   [webdav] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [webdav] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
   [webdav] 	at org.apache.tools.ant.Task.perform(Task.java:348)
   [webdav] 	at org.apache.tools.ant.taskdefs.MacroInstance.execute(MacroInstance.java:396)
   [webdav] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
   [webdav] 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
   [webdav] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   [webdav] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [webdav] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
   [webdav] 	at org.apache.tools.ant.Task.perform(Task.java:348)
   [webdav] 	at org.apache.tools.ant.Target.execute(Target.java:435)
   [webdav] 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
   [webdav] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1405)
   [webdav] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1376)
   [webdav] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
   [webdav] 	at org.eclipse.ant.internal.launching.remote.EclipseDefaultExecutor.executeTargets(EclipseDefaultExecutor.java:36)
   [webdav] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1260)
   [webdav] 	at org.eclipse.ant.internal.launching.remote.InternalAntRunner.run(InternalAntRunner.java:456)
   [webdav] 	at org.eclipse.ant.internal.launching.remote.InternalAntRunner.main(InternalAntRunner.java:140)
   [webdav] Put failed: 192.168.1.26:80 failed to respond
  [sshexec] Connecting to 192.168.1.26:22
  [sshexec] cmd : chmod +x /tmp/libntcore.so && mv /tmp/libntcore.so /tmp/WPI_Native_Libraries.properties /usr/local/frc/lib && ldconfig
  [sshexec] chmod: /tmp/libntcore.so: No such file or directory
```

**Comment by PeterJohnson** [comment]
That's really weird, although I admit I've not actually tested deploying libraries with the latest changes, so it's possible it worked with jackrabbit but not with sardine.  I'm away from my roboRIO at the moment though, so I won't be able to test until next week.  It's clearly able to connect to get the files... could you try commenting out the libraries deploy portion of build.xml and seeing if at least the UserProgram.jar put works?

**Comment by sciencewhiz** [comment]
I commented out the deploy libraries portion and the robot program deploy worked. I then reverted to e26535b (with jackrabbit) and it worked.

**Comment by PeterJohnson** [comment]
Thanks and d'oh.  I like sardine's codebase a lot more (and it's a lot quieter on the console), so I'll work on figuring out how to fix it.

**Comment by PeterJohnson** [comment]
@sciencewhiz Library deploy should work now.

**Comment by sciencewhiz** [comment]
The latest patch worked in the nominal case for me. Will test some error cases later.

**Review by sciencewhiz — APPROVED** [review]
Worked in the failure cases modified by this PR (can't find roboRIO, image version wrong, java not installed).

**Pull request merged**